### PR TITLE
Add defaults for entity records

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/26960

Fixes an issue with undefined values in `getMetadata`.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28851)

### CODE REVIEW

File: frontend/src/metabase-types/store/entities.ts [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
+  dashboards: Record<string, Dashboard>;
+  databases: Record<string, Database>;
+  schemas: Record<string, Schema>;
+  tables: Record<string, Table>;
+  fields: Record<string, Field>;
+  segments: Record<string, Segment>;
+  metrics: Record<string, Metric>;
+  snippets: Record<string, NativeQuerySnippet>;
+  users: Record<string, User>;
+  questions: Record<string, Card>;
```
Comment (ranquild) [inline frontend/src/metabase-types/store/entities.ts:28]: Technically an object cannot have a `number` key, so let's make it explicit.

**Comment by ranquild** [comment]
@deniskaber could you check that this change works for your PR?

File: frontend/src/metabase-types/store/entities.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase-types/store/entities.ts:28]: You can index it with a non-string key but it will be converted to a string

**Review by deniskaber — APPROVED** [review]
Works as expected. Thanks!

**Pull request merged**